tests/test_pytesseract.py

```diff
@@ -1,25 +1,246 @@
+# encoding: utf-8
+
 import os.path
+from multiprocessing import Pool
+from sys import platform, version_info
 
 import pytest
+from pytesseract import (
+    Output,
+    TSVNotSupported,
+    get_tesseract_version,
+    image_to_boxes,
+    image_to_data,
+    image_to_osd,
+    image_to_pdf_or_hocr,
+    image_to_string
+)
+from pytesseract.pytesseract import prepare
+
+try:
+    from PIL import Image
+except ImportError:
+    import Image
+
+try:
+    import pandas
+except ImportError:
+    pandas = None
+
+
+IS_PYTHON_2 = version_info[:1] < (3, )
+IS_PYTHON_3 = not IS_PYTHON_2
 
-from pytesseract import image_to_string
+TESSERACT_VERSION = tuple(get_tesseract_version().version)  # to skip tests
 
 DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
 
 
-@pytest.mark.pytesseract
-def test_quick_brown_dog_image():
-    test_file = os.path.join(DATA_DIR, 'test.png')
+pytestmark = pytest.mark.pytesseract  # used marker for the module
 
+
+@pytest.fixture(scope='session')
+def test_file():
+    return os.path.join(DATA_DIR, 'test.png')
+
+
+@pytest.mark.parametrize('test_file', [
+    # https://github.com/tesseract-ocr/tesseract/issues/2558
+    # os.path.join(DATA_DIR, 'test.bmp'),
+    # os.path.join(DATA_DIR, 'test.gif'),
+    os.path.join(DATA_DIR, 'test.jpg'),
+    Image.open(os.path.join(DATA_DIR, 'test.jpg')),
```

Comment:
Could probably simplify the logic if the `Image.open()` and `os.path.join` was moved into the test itself, and the only things parameterized was the file itself?

Reply:
I will split this tests in two different tests. You're right, I mixed it.

Reply:
`gif` images don't work with Tesseract 3, so I skip these cases.
`bmp` images don't work with any version of Tesseract.

Reply:
Hmm, interesting. According to the [Leptonica docs - Image I/O](http://tpgit.github.io/UnOfficialLeptDocs/leptonica/README.html#image-i-o), there is support for BMP and GIF.

Reply:
Also, as far as I can see from the Travis CI report, the `gif` test passes on **bionic**.
So indeed, the problem is the version of Leptopnica (leptonica-1.73 - xenial vs leptonica-1.75.3 - bionic which is also a bit old - Feb 16, 2018).